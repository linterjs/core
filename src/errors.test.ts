import {
  NoLinterProvidersError,
  LinterProviderLoadError,
  ModuleNotLinterProviderError
} from "./errors";

describe("Errors", () => {
  test("LinterProviderLoadError", () => {
    expect(
      new LinterProviderLoadError("@linter/provider-eslint")
    ).toMatchSnapshot();
  });

  test("ModuleNotLinterProviderError", () => {
    expect(
      new ModuleNotLinterProviderError("@linter/provider-eslint")
    ).toMatchSnapshot();
  });

  test("NoLinterProvidersError", () => {
    expect(new NoLinterProvidersError()).toMatchSnapshot();
  });
});
