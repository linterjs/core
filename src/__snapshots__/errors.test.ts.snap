// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errors LinterProviderLoadError 1`] = `[Error: There was a problem importing the linter provider module:  "@linter/provider-eslint".]`;

exports[`Errors ModuleNotLinterProviderError 1`] = `[Error: The imported module is not a linter provider:  "@linter/provider-eslint"]`;

exports[`Errors NoLinterProvidersError 1`] = `[Error: No linter providers found. Please install at least one linter provider package.]`;
